web3.js/src/transaction.ts

```diff
@@ -505,7 +512,7 @@ export class Transaction {
   /**
    * @internal
    */
-  _partialSign(message: Message, ...signers: Array<Account>) {
+  _partialSign(message: Message, ...signers: Array<TransactionSigner>) {
     const signData = message.serialize();
     signers.forEach(signer => {
       const signature = nacl.sign.detached(signData, signer.secretKey);
```

Comment:
Could wrap this guy in a `sign()` method on `TransactionSigner`

Reply:
Would love to do this, I stopped short of doing that in this PR because I wanted to maintain API compatibility and continue allowing apps to pass an `Account` into `sign(..)`. web3.js v2 should definitely have a signer interface more like the one in our Rust SDK which could be generic enough to allow signing via other wallets